scipy/optimize/_differentialevolution.py

```diff
@@ -768,15 +768,16 @@ def solve(self):
 
             # should the solver terminate?
             convergence = self.convergence
-
-            if (self.callback and
-                    self.callback(self._scale_parameters(self.population[0]),
-                                  convergence=self.tol / convergence) is True):
-
-                warning_flag = True
-                status_message = ('callback function requested stop early '
-                                  'by returning True')
-                break
+            if self.callback:
```

Comment:
Please make changes in accordance with my previous comment. This section is better written as:

```
if self.callback and self.callback(self.x, convergence=self.tol / convergence):
    warning_flag = True
    status_message = ('callback function requested stop early'
                                      ' by returning True')
    break
```
-or-
```
if self.callback:
    if self.callback(self.x, convergence=self.tol / convergence):
        warning_flag = True
        status_message = ('callback function requested stop early'
                                          ' by returning True')
        break
```
My preference is the former. @ilayn may prefer the second. The PR as it exists isn't really Pythonic.

Reply:
Nevermind, the first option is also my preference. Apparently I misread the line as 

```python
convergence=(self.tol / convergence is True)
```

This part of the code is really difficult to read.